src/async_doc_request.cpp

```diff
@@ -0,0 +1,144 @@
+#include "async_doc_request.h"
+#include "collection_manager.h"
+
+void AsyncDocRequestHandler::init(Store* store, int batch_interval, uint32_t db_size, uint32_t db_interval) {
+    async_req_store = store;
+    async_batch_interval = batch_interval;
+    async_db_size = db_size;
+    async_db_size_check_interval = db_interval;
+    last_batch_flush_secs = std::chrono::steady_clock::now();
+    last_db_size_check_secs = std::chrono::steady_clock::now();
+}
+
+void AsyncDocRequestHandler::check_handle_async_doc_request() {
+    auto now = std::chrono::steady_clock::now();
+    if (!async_request_batch.empty()) {
+        auto time_elapsed = std::chrono::duration_cast<std::chrono::seconds>(now - last_batch_flush_secs).count();
+        if (time_elapsed >= async_batch_interval) {
+            for (const auto& kv : async_request_batch) {
+                auto coll_name = kv.first;
+                auto collection = CollectionManager::get_instance().get_collection(coll_name);
+                auto async_reqs = kv.second;
+                nlohmann::json document;
+
+                std::vector <std::string> json_lines;
+                for (const auto& async_req: async_reqs) {
+                    json_lines.push_back(async_req.first);
+                }
+
+                nlohmann::json json_res = collection->add_many(json_lines, document, index_operation_t::CREATE, "",
```

Comment:
We can't always assume that all the writes will be of type `index_operation_t::CREATE`. We will have to see how to handle this. Definitely the stream of async writes can involve a mix of create, update and upsert.